### PR TITLE
added new way to use  api

### PR DESCRIPTION
**Pull request by ZeL1k7**
??

### CODE REVIEW

File: predict.py [hunk 1]
```diff
@@ -1,14 +1,27 @@
...
-    model = load_toxic_model(path='state_dict.pt')
-    with torch.no_grad():
-        logits = model(text)
-        probs = F.sigmoid(logits)
-    with open('prediction.json', 'w') as f:
-        json.dump({'logits': logits.item(), 'probalities': probs.item()}, f)
-    return 'prediction.json'
+class ToxicClassifer(Resource):
+
+    def get(self, file):
```
Comment (vladislavkruglikov) [inline predict.py]: Не очевидно, что file это Path или str, а назвал так как будто это file object.

File: predict.py [hunk 2]
```diff
@@ -1,14 +1,27 @@
...
+        with open(file) as f:  # получаем на вход json,в котором содержится наш текст
+            json_content = json.load(f)
```
Comment (vladislavkruglikov) [inline predict.py]: Просто data, не уложняй

File: predict.py [hunk 3]
```diff
@@ -1,14 +1,27 @@
...
+        text_list = json_content['text']
+
+        model = load_toxic_model(path='state_dict.pt')  # загружаем нашу модель и прогоняем через модель
```
Comment (vladislavkruglikov) [inline predict.py]: сделай 
```
def __init__(self, model: Optional[torch.nn?]):
    self.model = model
```

А саму model передавай в инит класса с помощью метода 
```
@lru(1)
def make_pt_model(path: Path) -> pt.model?:
   logger.info("Started loading model")
   model =  load_toxic_model(path='state_dict.pt')
   logger.info("Successfully loaded model")
  return model 
```

File: predict.py [hunk 4]
```diff
@@ -1,14 +1,27 @@
...
+        logits = []
+        probs = []
+
+        with torch.no_grad():
+            for string in text_list:
```
Comment (vladislavkruglikov) [inline predict.py]: for item in items

Comment (vladislavkruglikov) [reply predict.py]: А еще лучше с помощью pydantic определить класс item таким каким ты ожидаешь его видеть

from pedantic import BaseModel


class Item(BaseModel):
    id: int = Field(..., description="Unique item id")
    text: str = Field(..., description="какой то текст который ты там передаешь?")

File: predict.py [hunk 5]
```diff
@@ -1,14 +1,27 @@
...
+                logits.append(model(string).item())
```
Comment (vladislavkruglikov) [inline predict.py]: Большая вложеность

File: predict.py [hunk 6]
```diff
@@ -1,14 +1,27 @@
...
+                probs.append(F.sigmoid(logits).item())
+
+        with open('prediction.json', 'w') as f:  # сохраняем наши предсказания в json file
```
Comment (vladislavkruglikov) [inline predict.py]: вынеси названия файла в конструктор, и еще почему не w+? Если файла нету то у тебя не сможет записать в него

File: predict.py [hunk 7]
```diff
@@ -1,14 +1,27 @@
...
-    with torch.no_grad():
-        logits = model(text)
-        probs = F.sigmoid(logits)
-    with open('prediction.json', 'w') as f:
-        json.dump({'logits': logits.item(), 'probalities': probs.item()}, f)
-    return 'prediction.json'
+class ToxicClassifer(Resource):
+
+    def get(self, file):
+        with open(file) as f:  # получаем на вход json,в котором содержится наш текст
```
Comment (vladislavkruglikov) [inline predict.py]: as file было бы намного лучше, но у тебя open file почему то

File: predict.py [hunk 8]
```diff
@@ -1,14 +1,27 @@
...
+
+        with torch.no_grad():
+            for string in text_list:
+                logits.append(model(string).item())
+                probs.append(F.sigmoid(logits).item())
+
+        with open('prediction.json', 'w') as f:  # сохраняем наши предсказания в json file
+            json.dump({'logits': logits, 'probabilities': probs}, f)
+
+        return 'prediction.json', 200  # возвращаем названия json в случае успешного выполнения
```
Comment (vladislavkruglikov) [inline predict.py]: А в случае не успешного? Ты же не делаешь никаких try except, у тебя всегда будет либо 200 либо ничего так как упадет метод и ничего не вернется

File: predict.py [hunk 9]
```diff
@@ -1,14 +1,27 @@
...
+class ToxicClassifer(Resource):
+
+    def get(self, file):
+        with open(file) as f:  # получаем на вход json,в котором содержится наш текст
+            json_content = json.load(f)
+        text_list = json_content['text']
+
+        model = load_toxic_model(path='state_dict.pt')  # загружаем нашу модель и прогоняем через модель
+        logits = []
+        probs = []
```
Comment (vladislavkruglikov) [inline predict.py:17]: logits и probs? а какая разница? Я читаю и думаю что это одно и то же

**Review by vladislavkruglikov — COMMENTED** [review]
GLHF @ZeL1k7

File: predict.py (hunk 5 above)
Comment (ZeL1k7) [reply predict.py:21]: а как иначе?

File: predict.py (hunk 3 above)
Comment (ZeL1k7) [reply predict.py]: давно хотел закэшировать,спс

File: predict.py (hunk 5 above)
Comment (ZeL1k7) [reply predict.py]: а как иначе...

File: predict.py (hunk 6 above)
Comment (ZeL1k7) [reply predict.py]: не заметил,спасибо

File: predict.py [hunk 10]
```diff
@@ -1,14 +1,51 @@
...
+        self.model = self.make_pt_model('state_dict.pt')
+        self.output_filepath = 'predictions.json'
+    @lru_cache(1)  # загружаем модель + кэшируем ее
+    def make_pt_model(self,model_path):
+        print('Started loading model')
+        model = load_toxic_model(model_path)
+        print('Successful loaded model')
+        return model
+
+    def get(self, filepath):
```
Comment (vladislavkruglikov) [inline predict.py]: Смотри, в этом методе ты предсказываешь из json. Так назови predict_from_json, сразу станет очевидно, а что такое get не очень понятно. filepath замени на   path, тогда из метода predict_from_json сразу очевидно что path - json. ну и тайпинги `path: Path`, from pathlib import Path

**Review by vladislavkruglikov — COMMENTED** [review]
Оставил 1 коммент, остальное вроде норм

File: predict.py (hunk 5 above)
Comment (vladislavkruglikov) [reply predict.py]: Вроде не понятно модель у тебя на gpu или cpu, и тогда лучше было бы добавить `logits.append(model(string).cpu().item())`
Но как то много всего, я бы сделал
`
outputs = model(string).cpu().item()
logits.append(outputs)
`

File: predict.py (hunk 10 above)
Comment (ZeL1k7) [reply predict.py]: все тайпинги,аннотации и прочее будут будут,вопрос времени

File: model.py [hunk 11]
```diff
@@ -13,12 +12,18 @@ def __init__(self):
         self.tokenizer = AutoTokenizer.from_pretrained("DeepPavlov/rubert-base-cased")
         self.classifier = nn.Linear(768, 1)
 
-    def mean_pooling(self, model_output, attention_mask):
+    def mean_pooling(self, model_output: tuple(torch.FloatTensor), attention_mask: torch.FloatTensor) -> torch.FloatTensor:
+        """
+        Усредняет все вектора эмбеддингов в один вектор
```
Comment (vladislavkruglikov) [inline model.py:17]: Мне лично не нравятся комменты на русском, но дело вкуса
